### PR TITLE
Add a new ConfigSetter that accepts a callback function that will intercept HTTP calls

### PR DESCRIPTION
**Pull request by dnelson27**
This adds a new `ConfigSetter` function `WithHttpInterceptorAndHttpClientPtr` that allows clients to pass in a callback method that will be called for each HTTP request.

My organization currently uses `WithHttpClientPtr` to achieve this, but it would be great if the SDK supported it by default!

One significant use case is allowing clients to easily keep track of their API calls with their own observability code. 

### Example usage
```go
func main() {
	ctx, client, err := okta.NewClient(
		context.TODO(),
		okta.WithHttpInterceptorAndHttpClientPtr(
			func(r *http.Request) error {
				fmt.Println("Some observability code")
				return nil
			},
			http.DefaultClient,
			true,
		),
	)

	if err != nil {
		panic(err)
	}

	users, resp, err := client.User.ListUsers(ctx, nil)

	if err != nil {
		panic(err)
	}

	fmt.Println(users)
	fmt.Println(resp)
}
```

## Summary

Fixes #

## Type of PR

- [ ] Bug Fix (non-breaking fixes to existing functionality)
- [x] New Feature (non-breaking changes that add new functionality)
- [ ] Documentation update
- [ ] Test Updates
- [ ] Other (Please describe the type)

## Test Information

- [x] My PR required test updates 

Go Version: `go1.18.4 linux/amd64`
Os Version: `Linux 5.10.16.3-microsoft-standard-WSL2`
OpenAPI Spec Version:

## Signoff
- [x] I have submitted a CLA for this PR
- [x] Each commit message explains what the commit does
- [ ] I have updated documentation to explain what my PR does
- [x] My code is covered by tests if required
- [x] I ran `make fmt` on my code
- [x] I did not edit any automatically generated files

### CODE REVIEW

**Comment by dnelson27** [comment]
> Cool PR @dnelson27 , is it still WIP?

I just need to sign the CLA! Is that something I need to input somewhere or is checking the box my "Signature"?

I figure the code is sort of self-documenting, but I can add some docstrings if its unclear.

File: tests/unit/client_config_test.go [hunk 1]
```diff
@@ -90,3 +93,110 @@ func Test_will_error_if_private_key_authorization_type_with_missing_properties(t
...
+type InterceptingRoundTripperTest struct {
+	Name                    string
+	Blocking                bool
+	Interceptor             func(*http.Request) error
+	ExpectedTransportCalls  int
+	ExpectInterceptorCalled bool
+	ExpectSdkErrorThrown    bool
+}
+
+func Test_Intercepting_RounTtripper(t *testing.T) {
```
Comment (dnelson27) [inline tests/unit/client_config_test.go:106]: Typo, will fix

**Comment by monde** [comment]
> I just need to sign the CLA! Is that something I need to input somewhere or is checking the box my "Signature"?

@dnelson27 just follow the directions here
https://developer.okta.com/cla/

**Comment by dnelson27** [comment]
> > I just need to sign the CLA! Is that something I need to input somewhere or is checking the box my "Signature"?
> 
> @dnelson27 just follow the directions here https://developer.okta.com/cla/

Ah, I completely missed the section about the email on my first pass-over. Thanks!

**Comment by monde** [comment]
@dnelson27 if you could get that CLA in today that would be great, I'd like to release this PR and #372 / #369 as a single release.

**Comment by dnelson27** [comment]
> @dnelson27 if you could get that CLA in today that would be great, I'd like to release this PR and #372 / #369 as a single release.

CLA is submitted!

**Pull request closed without merging**